extensions/kafka-streams/deployment/src/main/java/io/quarkus/kafka/streams/deployment/KafkaStreamsProcessor.java

```diff
@@ -64,9 +64,6 @@ private void registerClassesThatAreLoadedThroughReflection(BuildProducer<Reflect
 
     private void registerCompulsoryClasses(BuildProducer<ReflectiveClassBuildItem> reflectiveClasses) {
         reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, false, StreamsPartitionAssignor.class));
-        // Class DefaultPartitionGrouper deprecated in Kafka 2.8.x and removed in 3.0.0
```

Comment:
The proper fix should be to verify if the class is there. The problem is that we still have users using Kafka 2.8.

Reply:
I updated the PR to check for the class' presence in the classpath instead.